pennylane/labs/phase_polynomials/parity_matrix.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Parity matrix representation"""
+
+from typing import Iterable
+
+import numpy as np
+
+import pennylane as qml
+
+
+def parity_matrix(circ: qml.tape.QuantumScript, wire_order: Iterable = None):
+    r"""
+    :doc:`Parity matrix intermediate representation <compilation/parity-matrix-intermediate-representation>` of a CNOT circuit
+
+    Args:
+        circ (qml.tape.QuantumScript): Quantum circuit containing only CNOT gates.
+        wire_order (Iterable): ``wire_order`` indicating how rows and columns should be ordered.
+
+    Returns:
+        np.ndarray: :math:`n \times n` Parity matrix
+
+    **Example**
+
+    .. code-block:: python
+
+        circ = qml.tape.QuantumScript([
+            qml.CNOT((3, 2)),
+            qml.CNOT((0, 2)),
+            qml.CNOT((2, 1)),
+            qml.CNOT((3, 2)),
+            qml.CNOT((3, 0)),
+            qml.CNOT((0, 2)),
+        ], [])
+
+        P = parity_matrix(circ, wire_order=range(4))
+
+    >>> P
+    array([[1., 0., 0., 1.],
+           [1., 1., 1., 1.],
+           [0., 0., 1., 1.],
+           [0., 0., 0., 1.]])
+
+    The corresponding circuit is the following.
+
+    .. code-block::
+
+        x_0: ────╭●───────╭X─╭●─┤  x_0 ⊕ x_3
+        x_1: ────│──╭X────│──│──┤  x_0 ⊕ x_1 ⊕ x_2 ⊕ x_3
+        x_2: ─╭X─╰X─╰●─╭X─│──╰X─┤  x_2 ⊕ x_3
+        x_3: ─╰●───────╰●─╰●────┤  x_3
+
+    """
+    wires = circ.wires
+
+    if wire_order is None:
+        wire_order = wires
+
+    if not qml.wires.Wires(wire_order).contains_wires(wires):
+        raise qml.wires.WireError(
+            f"The provided wire_order {wire_order} does not contain all wires of the circuit {wires}"
+        )
+
+    wire_map = {wire: idx for idx, wire in enumerate(wire_order)}
+
+    P = np.eye(len(wire_order))
+    for op in circ.operations:
+        control, target = op.wires
+        P[wire_map[target]] = P[wire_map[target]] + P[wire_map[control]]
```

Comment:
Does this work? I would suggest it not for Python code brevity, but because it resembles the nature of a CNOT "just adding the control to the target" more closely :) 
```suggestion
        P[wire_map[target]] += P[wire_map[control]]
```